### PR TITLE
diagnose: refine config diagnose result

### PR DESCRIPTION
**Pull request by crazycs520**
Signed-off-by: crazycs520 <crazycs520@gmail.com>

### What problem does this PR solve?

Remove the query and display the different config value directly.

before this PR:

```sql
+--------+------------------------------+------+----------+----------------+--------------+------------+----------+------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| RULE   | ITEM                         | TYPE | INSTANCE | STATUS_ADDRESS | VALUE        | REFERENCE  | SEVERITY | DETAILS                                                                                                                                                                                                                |
+--------+------------------------------+------+----------+----------------+--------------+------------+----------+------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| config | performance.run-auto-analyze | tidb |          |                | inconsistent | consistent | warning  | the cluster has different config value of performance.run-auto-analyze, execute the sql to see more detail: select * from information_schema.cluster_config where type='tidb' and `key`='performance.run-auto-analyze' |
+--------+------------------------------+------+----------+----------------+--------------+------------+----------+------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
```

This PR:

```sql
+--------+------------------------------+------+----------+----------------+--------------+------------+----------+---------------------------------------+
| RULE   | ITEM                         | TYPE | INSTANCE | STATUS_ADDRESS | VALUE        | REFERENCE  | SEVERITY | DETAILS                               |
+--------+------------------------------+------+----------+----------------+--------------+------------+----------+---------------------------------------+
| config | performance.run-auto-analyze | tidb |          |                | inconsistent | consistent | warning  | config value of 0.0.0.0:4001 is false |
|        |                              |      |          |                |              |            |          | config value of 0.0.0.0:4000 is true  |
+--------+------------------------------+------+----------+----------------+--------------+------------+----------+---------------------------------------+
```

### What is changed and how it works?



### Related changes

- Need to cherry-pick to the release branch

### Check List <!--REMOVE the items that are not applicable-->

Tests <!-- At least one of them must be included. -->

- Unit test

Side effects

- Breaking backward compatibility

### Release note <!-- bugfixes or new feature need a release note -->

- diagnose: refine config diagnose result

### CODE REVIEW

File: executor/inspection_result.go [hunk 1]
```diff
@@ -254,11 +254,33 @@ func (configInspection) inspectDiffConfig(_ context.Context, sctx sessionctx.Con
 		sctx.GetSessionVars().StmtCtx.AppendWarning(fmt.Errorf("check configuration consistency failed: %v", err))
 	}
 
+	generateDetail := func(tp, item string) string {
+		query := fmt.Sprintf("select value, instance from information_schema.cluster_config where type='%s' and `key`='%s';", tp, item)
+		fmt.Println(query)
```
Comment (crazycs520) [inline executor/inspection_result.go:259]:
```suggestion
```

File: executor/inspection_result.go [hunk 2]
```diff
@@ -254,11 +254,32 @@ func (configInspection) inspectDiffConfig(_ context.Context, sctx sessionctx.Con
...
+		}
+		m := make(map[string][]string)
+		for _, row := range rows {
+			value := row.GetString(0)
+			instance := row.GetString(1)
+			m[value] = append(m[value], instance)
+		}
+		groups := make([]string, 0, len(m))
+		for k, v := range m {
+			groups = append(groups, fmt.Sprintf("%s config value is %s", strings.Join(v, ","), k))
```
Comment (AilinKid) [inline executor/inspection_result.go:273]: should sort the `v`.([]string) first?   192.168.113.`1`, 192.168.113.`2`, 192.168.113.`3`...

Comment (crazycs520) [reply executor/inspection_result.go:274]: great.

**Pull request merged**